### DISCUSSION TITLE
Set Sound Settings (soundField)

### DISCUSSION THREAD

**Discussion by BurrPro** [jensrossbach/node-red-sony-audio#4] [Ideas & Requests]
Hello, just want to say thanks for the work you've put in with this node - I've got it working great, however I'm trying to configure the sound field settings on source change

I've got a music streamer box as well as an apple tv - I'm trying to build a flow that will change the sound fields between multi ch stereo and Dolby Surround when I change source?

I've got all the source change logic working, but I don't see an option inside the set sound settings command for changing between dolby and multi ch stereo.

Is there any way I can manually reference these options myself?

**Comment by jensrossbach** [comment]
Hello @BurrPro,
I'm happy that you like the nodes. Indeed I only added a few of the many available sound settings to the configuration page (actually only those that are supported by the Sony HT-XT3, the only Audio Control API capable device I have). If you can tell me, which sound setting exactly you need, I can add it to the menu. Alternatively, you can of course also directly send a `setSoundSettings` API to the device. This approach is described in the Wiki: [API Call](https://developer.sony.com/develop/audio-control-api/documentation/api-reference#_setsoundsettings_v1_1)

The API and the possible sound settings are described in the reference documentation of the Sony Audio Control API: [setSoundSettings](https://developer.sony.com/develop/audio-control-api/documentation/api-reference#_setsoundsettings_v1_1)

But as written above, if you tell me the required values for `settings.target` and `settings.value` of the `setSoundSettings` API, I can implement this in the node.

**Reply by BurrPro** [reply]
Thank you for the reply!

I'm wanting to be able to switch between A.F.D, Multi Ch Stereo, and Dolby

Does this help? This is the output of the getSoundSettings command


```
1: object
isAvailable: true
max: -1
min: -1
step: -1
title: "Multi Ch Stereo"
titleTextID: "sound-soundfield-multichstereo"
value: "multiChStereo"

3: object
isAvailable: true
max: -1
min: -1
step: -1
title: "A.F.D."
titleTextID: "sound-soundfield-audioformatdecoding"
value: "audioFormatDecoding"

4: object
isAvailable: true
max: -1
min: -1
step: -1
title: "Dolby Surround"
titleTextID: "sound-soundfield-dolbysurround"
value: "dolbySurround"
```

The API docs don't list these specific options, I've noticed with the STR-DN1080 there's a bit not documented.. 
If there's anything else you would need please let me know

**Reply by jensrossbach** [reply]
Would it be possible to provide the full JSON data you receive from the device for the `getSoundSettings` command? It would be good to see all candiate objects in order to know what are the possible values.

**Reply by BurrPro** [reply]
Of course, here you go :)

```
msg.payload : array[1]
array[1]

0: object
candidate: array[9]

0: object
isAvailable: true
max: -1
min: -1
step: -1
title: "2ch Stereo"
titleTextID: "sound-soundfield-afd-2ch"
value: "2chStereo"

1: object
isAvailable: true
max: -1
min: -1
step: -1
title: "Multi Ch Stereo"
titleTextID: "sound-soundfield-multichstereo"
value: "multiChStereo"

2: object
isAvailable: true
max: -1
min: -1
step: -1
title: "Direct"
titleTextID: "sound-soundfield-multi-direct"
value: "direct"

3: object
isAvailable: true
max: -1
min: -1
step: -1
title: "A.F.D."
titleTextID: "sound-soundfield-audioformatdecoding"
value: "audioFormatDecoding"

4: object
isAvailable: true
max: -1
min: -1
step: -1
title: "Dolby Surround"
titleTextID: "sound-soundfield-dolbysurround"
value: "dolbySurround"

5: object
isAvailable: true
max: -1
min: -1
step: -1
title: "Neural:X"
titleTextID: "sound-soundfield-neuralx"
value: "neuralX"

6: object
isAvailable: true
max: -1
min: -1
step: -1
title: "Front Surround"
titleTextID: "sound-soundfield-movie-frontsurround"
value: "frontSurround"

7: object
isAvailable: true
max: -1
min: -1
step: -1
title: "Audio Enhancer"
titleTextID: "sound-soundfield-music-audioenhancer"
value: "audioEnhancer"

8: object
isAvailable: false
max: -1
min: -1
step: -1
title: ""
titleTextID: "sound-soundfield-off"
value: "off"

currentValue: "multiChStereo"
deviceUIInfo: "soundFieldFig"
isAvailable: true
target: "soundField"
title: "Sound Field"
titleTextID: "sound-soundfield"
type: "enumTarget"
```

**Reply by jensrossbach** [reply]
Ok, I will investigate this, but I will not be able to test it as I don't have a device supporting these sound settings, so I need your help here.

**Reply by BurrPro** [reply]
Awesome, thanks a lot - I appreciate the help :)

I’m more than happy to help out

**Reply by jensrossbach** [reply]
While I'm adding the settings to the configuration UI, maybe you could test if they work using a direct API call with a control node.

Here are the steps to do:

1. Set the 'Action' to 'call API'.
2. Set the 'API' to a string containing `audio.setSoundSettings@1.1`.
3. Set the 'Parameters' to a JSON containing e.g.: `{"settings": [{"target": "soundField", "value": "multiChStereo"}]}`. For the `value` property, you can try out several values like "multiChStereo", "audioFormatDecoding" or "dolbySurround".

**Reply by BurrPro** [reply]
Hey @jensrossbach, so sorry it took so long to get around to doing this - calling the API manually works perfectly with all of the options the get audio settings call gave me, except for #8 - `titleTextID: "sound-soundfield-off"`.

**Reply by jensrossbach** [reply]
ok great, then I will prepare a release with the new soundfield options added.

**Reply by jensrossbach** [reply]
Version 2.0.1 is out which contains the new soundfield values.

**Reply by BurrPro** [reply]
Awesome, thank you very much for the support!!
